pkg/dns/dns.go

```diff
@@ -427,49 +431,43 @@ func (c *Client) unpatchCoreDNSConfig(deployment *appsv1.Deployment) (*corev1.Co
 
 // RestoreKubeDNS restores the KubeDNS configuration to pre-install state.
 func (c *Client) RestoreKubeDNS() error {
-	deployment, err := c.kubeClient.AppsV1().Deployments(metav1.NamespaceSystem).Get("kube-dns", metav1.GetOptions{})
+	kubeDNSDeployment, err := c.kubeClient.AppsV1().Deployments(metav1.NamespaceSystem).Get("kube-dns", metav1.GetOptions{})
 	if err != nil {
 		return err
 	}
 
 	// Get the currently loaded KubeDNS ConfigMap.
-	configMap, err := c.getKubeDNSConfigMap(deployment)
+	configMap, err := c.getConfigMap(kubeDNSDeployment.Namespace, "kube-dns")
```

Comment:
In most situation, assuming there will be a `kube-dns` ConfigMap in the KubeDNS deployment namespace is true. The restore logic for getting the ConfigMap is not the same as the patch logic. Not a blocker but I think it would be more robust to also using the iteration on volume logic here as well.